frontend/app/services/logout-timer-service.js

```diff
@@ -0,0 +1,45 @@
+import Service from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+
+export default class LogoutTimerService extends Service {
+  @tracked timeToLogoff;
+
+  AUTOLOGOFF_TIME = 298
+
+  logoutTimer = new Timer();
+
+  reset() {
+    if (this.logoutTimer.isRunning()){
+      this.logoutTimer = new Timer();
+    }
+  }
+
+  isRunning() {
+    return this.logoutTimer.isRunning()
+  }
+
+  start() {
+    this.logoutTimer.start()
+
+    this.logoutTimer.addEventListener('secondsUpdated', e => {
+      let passedTime = this.logoutTimer.getTotalTimeValues().seconds
+      this.calculateTimeToLogoff(passedTime)
+    })
+  }
+
+  redirectToNewSession() {
+    window.location.replace('/session/new');
```

Comment:
Does this also destroy the session? Or are we relying on the backend to do that?

Can you not just request DELETE /session and that redirects you directly?

Reply:
Sure. That would be more clever.